### PR TITLE
Use InstrumentedFunction instead of FunctionInfo where appropriate

### PR DESCRIPTION
**Pull request by dimitry-**
This change switches from FunctionInfo to InstrumentedFucntion in
CaptureData. It also does it for tracks and some other places.

LiveFunctionsDataView probably deserves another look, for now
it is still using FunctionInfo mostly because of selection/track
logic in DataManager.

Test: run orbit, instrument functions, take capture.

### CODE REVIEW

File: src/OrbitClientData/FunctionUtils.cpp [hunk 1]
```diff
@@ -35,7 +39,13 @@ uint64_t Offset(const FunctionInfo& func, const ModuleData& module) {
   return func.address() - module.load_bias();
 }
 
-bool IsOrbitFunc(const FunctionInfo& func) { return func.orbit_type() != FunctionInfo::kNone; }
+bool IsOrbitFunctionType(const FunctionInfo::OrbitType& type) {
```
Comment (dpallotti) [inline src/OrbitClientData/FunctionUtils.cpp:42]: I suggest `IsOrbitFunctionFromType`.

File: src/OrbitClientData/FunctionUtils.cpp [hunk 2]
```diff
@@ -35,7 +39,13 @@ uint64_t Offset(const FunctionInfo& func, const ModuleData& module) {
...
+  return type != FunctionInfo::kNone;
+}
+
+bool IsOrbitFunctionName(const std::string& function_name) {
```
Comment (dpallotti) [inline src/OrbitClientData/FunctionUtils.cpp:46]: I suggest `IsOrbitFunctionFromName`.

File: src/OrbitClientModel/CaptureData.cpp [hunk 3]
```diff
@@ -143,6 +146,17 @@ std::optional<uint64_t> CaptureData::FindFunctionAbsoluteAddressByAddress(
   }
   return std::nullopt;
 }
+const FunctionInfo* CaptureData::FindFunctionByModulePathAndOffset(const std::string& module_path,
```
Comment (dpallotti) [inline src/OrbitClientModel/CaptureData.cpp:149]:
```suggestion
}

const FunctionInfo* CaptureData::FindFunctionByModulePathAndOffset(const std::string& module_path,
```

**Review by dpallotti — APPROVED** [review]
To me everything looks good. I can imagine this will need quite a bit of manual testing though. I would at least have another pair of eyes to take a look.

File: src/OrbitClientData/FunctionUtils.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitClientData/FunctionUtils.cpp:42]: done.

File: src/OrbitClientData/FunctionUtils.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitClientData/FunctionUtils.cpp:46]: done.

File: src/OrbitClientModel/CaptureData.cpp (hunk 3 above)
Comment (dimitry-) [reply src/OrbitClientModel/CaptureData.cpp:149]: done.

File: src/OrbitClientModel/CaptureDeserializerTest.cpp [hunk 4]
```diff
@@ -221,12 +223,12 @@ TEST(CaptureDeserializer, LoadCaptureInfoOnCaptureStarted) {
...
-        FunctionInfo actual_function_info =
+        const InstrumentedFunction& actual_function_info =
             actual_instrumented_functions.at(kInstrumentedFunctionId);
 
-        EXPECT_EQ(actual_function_info.name(), instrumented_function.name());
-        EXPECT_EQ(actual_function_info.pretty_name(), instrumented_function.pretty_name());
-        EXPECT_EQ(actual_function_info.size(), instrumented_function.size());
+        EXPECT_EQ(actual_function_info.function_name(), instrumented_function.pretty_name());
+        EXPECT_EQ(actual_function_info.file_path(), instrumented_function.loaded_module_path());
+        EXPECT_EQ(actual_function_info.file_offset(), instrumented_function.address() - load_bias);
```
Comment (pierricgimmig) [inline src/OrbitClientModel/CaptureDeserializerTest.cpp:231]: was the check for ".size()" removed on purpose?

**Review by pierricgimmig — APPROVED** [review]
LGTM, thanks!

File: src/OrbitClientModel/CaptureDeserializerTest.cpp (hunk 4 above)
Comment (dimitry-) [reply src/OrbitClientModel/CaptureDeserializerTest.cpp:231]: Yes, InstrumentedFunction message does not have size.

**Pull request merged**